test/index.js

```diff
@@ -212,4 +212,28 @@ describe('Errors', () => {
 
         stream.end({ x: undefined });
     });
+
+    it('accepts objects with circular dependencies', (done) => {
+
+        const stream = new Errors({});
+
+        stream.on('readable', () => {
+
+            const result = stream.read();
+
+            if (!result) {
+                return done();
+            }
+
+            expect(result.data.a).to.equal(1);
+            expect(result.data.b.c).to.equal('x');
+            expect(result.data.b.d.stringifyable).to.equal({ name: 'Error', stack: 'Some\nstack', message: 'foo' });
+        });
+
+        const err = new Error('foo');
+        err.stack = 'Some\nstack';
+        const data = { a: 1, b: { c: 'x', d: err } };
+        data.b.circ = data;
```

Comment:
We should probably add test for `result.data.b.circ` to at least be explicit about expected behavior.

Reply:
I removed the irrelevant parts from the tests, but with regards to what this is testing, the code is not removing or replacing the circular ref or anything like date, so all this test is checking for is that the object is correctly scanned for errors even if it has circular references in and that it doesn't break.